### ISSUE TITLE
`generateAppearancesIfNeeded` does not allow enabled values other than "/Yes" for checkboxes

### ISSUE THREAD

**Issue by dmjohnsson23** [qpdf/qpdf#1056]
The `QPDFAcroFormDocumentHelper::generateAppearancesIfNeeded` (or, more accurately, `QPDFFormFieldObjectHelper::setCheckBoxValue`) method does not allow enabled values other than "/Yes". However, this does not work for all PDFs. For example, [here](https://www.va.gov/vaforms/va/pdf/VA9.pdf) is a PDF that uses "/1" as the enabled value. Checkboxes should be allowed to be set to any value in the "/AP" dictionary, similar to how radio buttons work.

I was able to track the issues to these two lines, but I'm not a C programmer, so I thought it best to let someone else make the actual changes. (I'm using QPDF by way of PikePdf, rather than directly)
* https://github.com/qpdf/qpdf/blob/81823f4032caefd1050bccb207d315839c1c48db/libqpdf/QPDFFormFieldObjectHelper.cc#L307
* https://github.com/qpdf/qpdf/blob/81823f4032caefd1050bccb207d315839c1c48db/libqpdf/QPDFFormFieldObjectHelper.cc#L419

I've never encountered a checkbox that used anything other than "/Off" for the disabled value, but the enabled value appears to be a bit a a wild west.

**Comment by jberkenbilt** [comment]
Thanks -- I should be able to work this in.

**Comment by jberkenbilt** [comment]
@dmjohnsson23 Thanks for your analysis. This made my work very easy. I verified in the spec as well -- the old functionality was incorrect. Thanks for taking the time to report this. It will be fixed for 11.7.0.

**Comment by amargerandhw** [comment]
Hello @jberkenbilt, thanks for your awesome project !
I think the fix done here https://github.com/qpdf/qpdf/pull/1089 is not complete as the method `QPDFFormFieldObjectHelper::setCheckBoxValue` still set `/Yes` when you pass true. I think it should be the corresponding true value of the checkbox.
If I'm not mistaken that could be done by checking the keys returned by `AP.getKey("/N").getKeys()` ?

Thank you in advance

**Comment by dmjohnsson23** [comment]
This may or may not be useful to you, but here is how I ended up working around this bug to set the checkbox appearance:

https://github.com/dmjohnsson23/pdform/blob/432dc1474d552e1eea1f07dad8c729e22e0ea610/pdform/model/form_field.py#L240

You're more than welcome to copy anything you like from my code above. I also have some code for handling multi-line text fields that may be of interest (though not relevant to this issue of course).

Basically, I'm using the first value I find from the AP.N dictionary that isn't /Off. This seems to be correct based on my own reading of the (admittedly very difficult to understand) spec, as well as my own experience. Section12.7.4.2.3 says, "The appearance for the off state is optional but, if present, shall be stored in the appearance dictionary under the name Off"
